controller/state.go

```diff
@@ -179,20 +180,23 @@ func (m *appStateManager) GetRepoObjs(app *v1alpha1.Application, sources []v1alp
 	// revisions for the rollback
 	refSources, err := argo.GetRefSources(context.Background(), sources, app.Spec.Project, m.db.GetRepository, revisions, rollback)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to get ref sources: %w", err)
+		return nil, nil, false, fmt.Errorf("failed to get ref sources: %w", err)
 	}
 
+	// by default should be for regular sync, if application has no AnnotationKeyManifestGeneratePaths , it should work as before
+	revisionUpdated := true
```

Comment:
After this point, nothing is setting the boolean back to false 🤔 so it always returns true, except during error. It looks like this return variable is not necessary.

Reply:
Yeah, i changed default to true, because it always returned false before for apps that has no AnnotationKeyManifestGeneratePaths annotation. I am reverting it back